File: lib/core/models/banner_model.dart
class HomeBanner {
  final String id;
  final String title;
  final String subtitle;
  final String image;
  final String targetType;
  final String targetValue;

  const HomeBanner({
    required this.id,
    required this.title,
    required this.subtitle,
    required this.image,
    required this.targetType,
    required this.targetValue,
  });

  factory HomeBanner.fromJson(Map<String, dynamic> json) {
    return HomeBanner(
      id: (json["_id"] ?? json["id"] ?? "").toString(),
      title: (json["title"] ?? "").toString(),
      subtitle: (json["subtitle"] ?? "").toString(),
      image: (json["image"] ?? "").toString(),
      targetType: (json["targetType"] ?? "").toString(),
      targetValue: (json["targetValue"] ?? "").toString(),
    );
  }
}

File: src/modules/banners/banner.model.js
const mongoose = require("mongoose");

const bannerSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    subtitle: {
      type: String,
      trim: true,
      default: "",
    },
    image: {
      type: String,
      required: true,
      trim: true,
    },
    targetType: {
      type: String,
      enum: ["category", "product", "url"],
      required: true,
    },
    targetValue: {
      type: String,
      required: true,
      trim: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
    sortOrder: {
      type: Number,
      default: 0,
    },
  },
  {
    timestamps: true,
  }
);

const Banner = mongoose.model("Banner", bannerSchema);

module.exports = Banner;
